### ISSUE TITLE
Fiona 1.8b: warnings about "driver GPKG does not support creation option ENCODING"

### ISSUE THREAD

**Issue by jorisvandenbossche** [Toblerity/Fiona#664]
So this is a bit similar to https://github.com/Toblerity/Fiona/issues/627 and not sure I should report it, but, using the latest Fiona and writing a simple GPKG file, I get:

```
In [1]: import geopandas                                                                                                                                                                                            

In [4]: from shapely.geometry import Point                                                                                                                                                                          

In [5]: df = geopandas.GeoDataFrame({'a': [1, 2, 3], 'b': [True, False, True], 'geometry': [Point(0, 0), Point(1, 1), Point(2, 2)]})                                                                                

In [11]: df.to_file("__test_bool.gpkg", driver='GPKG')                                                                                                                                                              
CPLE_NotSupported in driver GPKG does not support creation option ENCODING
CPLE_NotSupported in dataset /home/joris/scipy/__test_bool.gpkg does not support layer creation option ENCODING
CPLE_NotSupported in dataset /home/joris/scipy/__test_bool.gpkg does not support layer creation option ENCODING

In [12]: fiona.__version__                                                                                                                                                                                          
Out[12]: '1.8b1'
```

Those warnings were already logged in Fiona 1.7.x, but now they are shown in the console making them much more prominent, and potentially confusing for the user. 

For this specific warning, is there a way to avoid it? (in fiona or geopandas)

**Comment by sgillies** [comment]
@jorisvandenbossche passing `VALIDATE_OPEN_OPTIONS=NO` when you call `fiona.open()` will suppress this. See https://www.gdal.org/classGDALDataset.html#a9cb8585d0b3c16726b08e25bcc94274a for reference.

**Comment by jorisvandenbossche** [comment]
I don't get this to work. I am trying to pass `fiona.open(...., VALIDATE_OPEN_OPTIONS='NO')`, is that the correct thing? Also tried with `False` instead of `'NO'` and also tried passing it to `fiona.Env`. 

A pure fiona example:

```
import fiona                                                                                                                                                           

schema = {'geometry': 'Point', 'properties': {'a': 'int'}}                                                                                                             
features = [{'id': '0', 
  'type': 'Feature', 
  'properties': {'a': 1}, 
  'geometry': {'type': 'Point', 'coordinates': (0.0, 0.0)}}, 
 {'id': '1', 
  'type': 'Feature', 
  'properties': {'a': 2}, 
  'geometry': {'type': 'Point', 'coordinates': (1.0, 1.0)}}, 
]                                                                                                                                                                      

with fiona.Env(): 
    with fiona.open('__test_encoding.geojson', 'w', driver="GeoJSON", schema=schema) as dst: 
        dst.writerecords(features)                                                                                                                                     
```

gives the warning whatever combination I try of passing the option:

```
In [47]: with fiona.Env(VALIDATE_OPEN_OPTIONS=False): 
    ...:     with fiona.open('__test_encoding.geojson', 'w', driver="GeoJSON", schema=schema, VALIDATE_OPEN_OPTIONS=False) as dst: 
    ...:         dst.writerecords(features)                                                                                                                                     
CPLE_NotSupported in driver GeoJSON does not support creation option ENCODING
CPLE_NotSupported in dataset __test_encoding.geojson does not support layer creation option ENCODING
CPLE_NotSupported in dataset __test_encoding.geojson does not support layer creation option ENCODING
```

**Comment by sgillies** [comment]
Confirmed @jorisvandenbossche . I'm going to let this one slide to the next bug fix release, 1.8.1.

**Comment by jorisvandenbossche** [comment]
Sure, this certainly does not need to hold up the 1.8.0 release.

To be clear, I am not arguing to not show warnings in general. But, given that the GDAL warnings are now always shown to the user, I think we should try harder to suppress warnings if possible / desired. 

In this case, the warning is not shown because of an (invalid) option the user passed, but from a default option of fiona that is passed through to a driver that does not support it. 
Writing any valid GeoJSON or Geopackage file with fiona will show the warning by default.

---

I personally would think making sure the warning is not shown would fit in fiona (but not going to make hard claims as I am not that familiar with the code base). But, if I want to hide them from the geopandas side (assuming we get the VALIDATE_OPEN_OPTIONS working), do you think there can be any harm in doing that? Since it is a general option to not validate options, we might be hiding other, more relevant warnings?
